share/functions/fish_fossil_prompt.fish

```diff
@@ -0,0 +1,53 @@
+function fish_fossil_prompt --description 'Write out the fossil prompt'
+    # Bail if fossil is not available
+    if not command -sq fossil
+        return 1
+    end
+
+    # Bail if not a fossil checkout
+    if not test -e .fslckout
+        return 127
+    end
+
+    # Parse fossil info
+    set -l fossil_info (fossil info)
+    string match --regex --quiet \
+        '^project-name:\s*(?<fossil_project_name>.*)$' $fossil_info
+    string match --regex --quiet \
+        '^tags:\s*(?<fossil_tags>.*)$' $fossil_info
+
+    echo -n '['
+    set_color --bold magenta
+    echo -n $fossil_project_name
+    set_color normal
+    echo -n ':'
+    set_color --bold yellow
+    echo -n $fossil_tags
+    set_color normal
+
+    # Parse fossil status
+    set -l fossil_status (fossil status)
+    if string match --index --regex --quiet '^ADDED' $fossil_status
```

Comment:
I don't know what the `--index` is for, it seems to not do anything.

Reply:
You can also use glob-mode here:

```fish
if string match --quiet 'ADDED*' $fossil_status
```

which is a bit shorter.

Reply:
changed